src/Categories/Bicategory/Construction/LaxSlice.agda

```diff
@@ -0,0 +1,337 @@
+{-# OPTIONS --without-K --safe #-}
+
+-- Mentioned in passing https://ncatlab.org/nlab/show/slice+2-category
+
+open import Categories.Bicategory
+
+module Categories.Bicategory.Construction.LaxSlice
+       {o ℓ e t}
+       (𝒞 : Bicategory o ℓ e t)
+       where
+
+open import Categories.Enriched.Category
+open import Categories.Category renaming (Category to 1Category)
+open import Categories.Morphism.Reasoning
+
+open import Categories.Bicategory.Extras 𝒞
+
+open import Level
+
+record SliceObj (X : Obj) : Set (t ⊔ o) where
+  constructor sliceobj
+  field
+    {Y} : Obj
+    arr : Y ⇒₁ X
+
+module SliceHom (A : Obj) where
+
+  record Slice⇒₁ (X Y : SliceObj A) : Set (o ⊔ ℓ) where
+    constructor slicearr₁
+    private
+      module X = SliceObj X
+      module Y = SliceObj Y
+
+    field
+      {h} : X.Y ⇒₁ Y.Y
+      Δ   : X.arr ⇒₂ (Y.arr ∘ₕ h)
+
+  record Slice⇒₂ {X Y : SliceObj A} (H K : Slice⇒₁ X Y) : Set (ℓ ⊔ e) where
+    constructor slicearr₂
+    private
+      module Y = SliceObj Y
+      module H = Slice⇒₁ H
+      module K = Slice⇒₁ K
+
+    field
+      {ϕ} : H.h ⇒₂ K.h
+      E   : K.Δ ≈ (Y.arr ▷ ϕ ∘ᵥ H.Δ)
+
+  open hom.Equiv
+  open import Categories.Functor using (Functor)
+
+  _∘'_ : ∀ {X Y : SliceObj A}{H K L : Slice⇒₁ X Y} → Slice⇒₂ K L → Slice⇒₂ H K → Slice⇒₂ H L
+  _∘'_ {X}{Y}{H}{K}{L} (slicearr₂ {ϕ = ϕ} E) (slicearr₂ {ϕ = ψ} F) = slicearr₂ {ϕ = ϕ ∘ᵥ ψ} 
+     (begin
+      L.Δ ≈⟨ E ⟩
+      (Y.arr ▷ ϕ ∘ᵥ K.Δ) ≈⟨ refl⟩∘⟨ F ⟩
+      Y.arr ▷ ϕ ∘ᵥ (Y.arr ▷ ψ ∘ᵥ H.Δ) ≈˘⟨ hom.assoc ⟩
+      (Y.arr ▷ ϕ ∘ᵥ Y.arr ▷ ψ) ∘ᵥ H.Δ ≈⟨ (∘ᵥ-distr-▷ ⟩∘⟨refl) ⟩
+      Y.arr ▷ (ϕ ∘ᵥ ψ) ∘ᵥ H.Δ ∎
+        )
+    where module X = SliceObj X
+          module Y = SliceObj Y
+          module H = Slice⇒₁ H
+          module K = Slice⇒₁ K
+          module L = Slice⇒₁ L
+          open 1Category (hom X.Y A)
+          open HomReasoning
+          open Equiv
+          open import Categories.Morphism.Reasoning (hom X.Y A)
+          open import Relation.Binary.Core using (Rel)
+          open import Function.Base using (_$_)
+          
+  open SliceObj
+  SliceHomCat : SliceObj A → SliceObj A → 1Category (o ⊔ ℓ) (ℓ ⊔ e) e
+  SliceHomCat X Y = record
+    { Obj = Slice⇒₁ X Y
+    ; _⇒_ = Slice⇒₂
+    ; _≈_ = λ where
+      (slicearr₂ {ϕ} _) (slicearr₂ {ψ} _) → ϕ ≈ ψ 
```

Comment:
As you end up using the implicits in slicearr all the time, I think that says that they should actually be explicit.

Reply:
Seems fair. I based this on the analogous constructor in the slice 1-category library. It works well when using the constructor because the morphism can be inferred from the equality proof.

Reply:
Note that this isn't a "for sure" thing. You need to differentiate between needing to refer to it explicitly in fundamental definitions versus in applications. If it's seldom needed in applications, then leaving it implicit might be fine.